addons/binding/org.openhab.binding.konnected/src/main/java/org/openhab/binding/konnected/handler/KonnectedHandler.java

```diff
@@ -0,0 +1,476 @@
+                    payload.addActuators(module);
+                    StringType state = new StringType("0");
+                    updateState(channelId, state);
+                    logger.trace("Channel {} is of type Switch", channel.toString());
+                } // if the itemType is dimensionless then this is the humididty of the dht22, which we add in the Temp,
+                  // so we don't add anything here
+                if (itemType.equalsIgnoreCase("Number:Dimensionless")) {
+                    logger.trace("Channel {} is of type Number:Dimensionless", channel.toString());
+                } // if the itemType is Number:Temperature then this is a Temp Sensor
+                if (itemType.equalsIgnoreCase("Number:Temperature")) {
+                    logger.trace("Channel {} is of type Number:Temperature", channel.toString());
+                    Configuration configuration = channel.getConfiguration();
+                    if (configuration.get("pollinterval") == null) {
+                        module.setPollInterval(3);
+                    } else {
+                        module.setPollInterval(((BigDecimal) configuration.get("pollinterval")).intValue());
+                    }
+                    logger.trace("The Temperature Sensor Type is: ", configuration.get("tempsensorType").toString());
+                    if ((boolean) configuration.get("tempsensorType")) {
+                        // add it as a dht22 module
+                        payload.addDht22(module);
+                    } else {
+                        // add to payload as a DS18B20 module if the parameter is false
+                        payload.addDs18b20(module);
+                    }
+                } else {
+                    logger.debug("Channel {} is of type {} which is not supported by the konnected binding",
+                            channel.toString(), itemType);
+                }
+            } else {
+                logger.debug("The Channel {} is not linked to an item", channel.getUID());
+            }
+        });
+        // Create Json to Send to Konnected Module
+
+        String payloadString = gson.toJson(payload);
+        logger.debug("The payload is: {}", payloadString);
+        return payloadString;
+    }
+
+    /**
+     * Obtains the configured hostname of the openHAB server to send to the moduele.
+     *
+     * @return either the ipaddress obtained by the runtime or defined by the user in the thing configuration
+     * @throws UnknownHostException if no host defined
+     */
+    protected String getHostName() throws UnknownHostException {
+        // returns the local address of the openHAB server from the NetworkAddressService,
+        // or uses the configured path from the thing if the user adds one
+        if (config.hostAddress == null) {
+            return callbackIpAddress;
+        } else {
+            logger.debug("User has provided an Ip address on the thing for the openhab host of : {}",
```

Comment:
Ip -> ip
openhab -> openHAB

Reply:
this section has been removed as the host address is no longer configurable on the thing per https://github.com/openhab/openhab2-addons/pull/3702#discussion_r232198442